apps/game_backend/lib/game_backend/curse_of_mirra/quests.ex

```diff
@@ -298,6 +296,79 @@ defmodule GameBackend.CurseOfMirra.Quests do
     |> Repo.transaction()
   end
 
+  def get_user_quest_progress(%UserQuest{quest: %Quest{} = quest} = user_quest, arena_match_results) do
+    arena_match_results =
+      case quest.type do
+        type when type in ["daily", :daily] ->
+          Enum.filter(arena_match_results, fn arena_match_result ->
```

Comment:
```suggestion
     if quest.type in ["daily", :daily] do
        Enum.filter(arena_match_results, fn arena_match_result ->
```

seems more simple. Also, why do we need to handle both string and atom cases?

And why just the dailies? What about weekly or bounty ones?

Reply:
1. I don't actually remember why we handle both cases, will see if it breaks under any circumstance
2. Because two reasons
   a. Users don't get to complete bounties manually, that is completed inside a match and we have a proper check in arena where we don't need to filter the match result since it only checks the one in the match
   b. This method will be used only inside the `get_user_by_id_and_game_id` method that has the `arena_match_results_preloads` which only preloads matches that happened in that week so for "weeklies" you don't need that filter. It was the easiest solution i found to not perform a query for each user quest that we want to check

Reply:
2. Alright.
What about 1? Do we need both checks? Can we fix that somehow to use string or atom but not both?
Finally, did you see my "use if not case" suggestion? I believe is cleaner.

Reply:
1. Yeah the string case wasn't necessary at all, since this function is used in two places
   a. to get the quest progress and send it to the client to display
   b. to check if the user has completed the quest to be claimed, and both cases uses atoms 

I think this was a trace from a previews implementation so i just removed it